### ISSUE TITLE
Improve Asset Storage Backend Connection Tracking 

### ISSUE THREAD

**Issue by anthony-sarkis** [diffgram/diffgram#861]
### Context
In the past we operated under the assumption of a single storage backend
so it was rare that (from a system level) a blob could ever inaccessible, missing or unavailable

Recent changes, and directional ideas, make it more possible
* As we add more storage backends
  *  it's possible a file could be in one backend, and if the user switches it will become missing
in the case of minio and local dev for example
* As we grow embedded support and users more deeply integrate Diffgram 
* End user managed storage could be disrupted. e.g. user deletes bucket
* New end user goals / life cycle rules, e.g guarantee delete after x period of time.
* As we explore more IAM and alternative storage concepts, it becomes more likely 
that a blob will be auth blocked, inaccessible or deleted from some other process

So basically before the assumption was if the blob was missing it was "really bad"
Now there could be valid, even expected, reasons.
So we want to shift more into software managed and reduce expectations about having the same storage always available.

### Ideas
* Add better 404 / error handling if the bob / file path is not found (since this could always happen in theory)
* General shift to no longer assume a blob will be available
* Capture more information about the where the resource lives
for example, if we also shift to making the "backend" itself a `connection` object,
this might be as easy as storing the connection ID.

Semi related, this could be a step towards cases where we want to keep Diffgram generated data artifacts in the same place
as the original data.

**Comment by anthony-sarkis** [comment]
Another example of this is on the frontend UI, display a smoother message indicating the blob is currently unavailable
